### DISCUSSION TITLE
None identical results from differents run

### DISCUSSION THREAD

**Discussion by remilamoureuxlevesque** [yzhao062/pyod#525] [Q&A]
I am running a benchmark for evaluating the best algorithm to use for my application, but I cannot get twice the same result from running the benchmark two times in a row.

Does it have somethings in the algorithm that could be related to that ? or are you able to get repetitive results from algorithms in pyOD ?

Thanks for you feedback.

**Comment by yzhao062** [comment]
Many od algorithms have built in randomness

**Comment by remilamoureuxlevesque** [comment]
Oh ok, 

Is there a way to remove or reduce the effect of the randomness ?

**Comment by yzhao062** [comment]
it depends on algorithms, knn lof is deterministic. isolation forest is not

**Comment by remilamoureuxlevesque** [comment]
Ok thank you for your anwser, I will look into that.
